### PR TITLE
sts: Add STS support for S3 (PROJQUAY-6362)

### PR DESCRIPTION
**Pull request by jonathankingfc**
- Add STS authentication to s3
- Add STS validation to config tool
- PR automatically refreshes token on expiry

### CODE REVIEW

File: config-tool/pkg/lib/shared/storage_validators.go [hunk 1]
```diff
@@ -143,6 +144,58 @@ func ValidateStorage(opts Options, storageName string, storageType string, args
...
+		roleArn := args.STSRoleArn
+		sess := session.Must(session.NewSession(&aws.Config{
+			Credentials: awscredentials.NewStaticCredentials(args.STSUserAccessKey, args.STSUserSecretKey, ""),
+		}))
+		svc := sts.New(sess)
+		roleToAssumeArn := roleArn
+		durationSeconds := int64(3600)
+		assumeRoleInput := &sts.AssumeRoleInput{
+			RoleArn:         aws.String(roleToAssumeArn),
+			RoleSessionName: aws.String("quay"),
```
Comment (arborite-rh) [inline config-tool/pkg/lib/shared/storage_validators.go:163]: is this documented anywhere as the session name?

**Comment by dmage** [comment]
@jonathankingfc can you write a config example and brief instructions on how to use it?

**Comment by LiZhang19817** [comment]
What's the time interval that Quay 3.11 will refresh the STS Token?

**Comment by jonathankingfc** [comment]
@LiZhang19817 It will automatically be refreshed upon expiration. That can be set by the AWS admin

File: config-tool/pkg/lib/shared/storage_validators.go (hunk 1 above)
Comment (jonathankingfc) [reply config-tool/pkg/lib/shared/storage_validators.go:163]: This session name can be anything

Comment (arborite-rh) [reply config-tool/pkg/lib/shared/storage_validators.go:163]: yes, understood, and it should be fairly obvious from the name of the session, but some customers under more regulatory requirements may want to see documentation of the STS token session retrieval

**Comment by jonathankingfc** [comment]
@dmage @LiZhang19817 I've added documentation on how to set up and configure STS with Quay. It can be found at this document 

https://docs.google.com/document/d/1hvBwtsggRZueMcE7-VQJ89YK3RYHJ13QZZxAyCw7M48

File: storage/cloud.py [hunk 2]
```diff
@@ -1169,3 +1183,41 @@ def _load_private_key(self, cloudfront_privatekey_filename):
...
+class STSS3Storage(S3Storage):
+    def __init__(
+        self,
+        context,
+        storage_path,
+        s3_bucket,
+        sts_role_arn=None,
+        sts_user_access_key=None,
+        sts_user_secret_key=None,
+        s3_region=None,
```
Comment (LiZhang19817) [inline storage/cloud.py:1197]: If not specifying the aws region, will quay call AWS STS global endpoint or regional endpoint?

**Pull request merged**